eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -147,6 +147,30 @@ default T getOnly()
 
         throw new IllegalStateException("Size must be 1 but was " + this.size());
     }
+    
+    /**
+     * Returns the element as an optional if the iterable has exactly one element.
+     * If the iterable is empty {@link Optional#empty} is returned.
+     * Otherwise, throw {@link IllegalStateException}.
+     *
+     * @return an element of an iterable wrapped in an optional or an empty optional.
+     * @throws IllegalStateException if iterable has more than one element.
```

Comment:
Thanks for the feedback!

I can not claim the same level of experience with `getOnly`, as I was just looking for an easy to contribute issue.  

I think `getOnlyOptional` returning `Optional.empty()` for multiple elements can be confusing for these reasons:
* difference to `getOnly()`
* difference to other popular implementations? ([Iterables.getOnlyElement()](https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Iterables.html#getOnlyElement-java.lang.Iterable-) or [MoreCollectors.toOptional()](https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/MoreCollectors.html#toOptional--) from guava throw an `IllegalArgumentException` for multiple elements)
* three meanings of  `Optional.empty()`: empty iterable, iterable with multiple elements or iterable containing a single `Null` element

Of course I'm fine with changing the implementation to not throw if that is decided. :)